### PR TITLE
Command line arguments and GTest integration

### PR DESCRIPTION
**Pull request by dlangbe**
Added the ability to parse the following command line arguments for hipTensor tests:
```
-y {Specify input YAML filename}
-o {Specify output filename} (currently unused)
```

Added the following tests with their own default parameter configuration YAML files:
```
bilinear_contraction_f32_test
bilinear_contraction_f64_test
scale_contraction_f32_test
scale_contraction_f64_test
```

Added initial GTest integration for running parameterized test suites.

### CODE REVIEW

**Review by mkarunan — APPROVED** [review]

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -27,6 +27,26 @@
 # Find / configure LLVM
 add_subdirectory(llvm)
 
+include( FetchContent )
+
+FetchContent_Declare(
+  googletest
+  GIT_REPOSITORY https://github.com/google/googletest.git
+  GIT_TAG 609281088cfefc76f9d0ce82e1ff6c30cc3591e5
```
Comment (cgmillette) [inline test/CMakeLists.txt]: Can we upgrade this GIT_TAG to  **release-1.12.1**

File: test/llvm/hiptensor_options.cpp [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <llvm/Support/CommandLine.h>
+
+#include "hiptensor_options.hpp"
+
+// Get input/output file names
+llvm::cl::OptionCategory   HiptensorCategory("hipTensor Options",
```
Comment (cgmillette) [inline test/llvm/hiptensor_options.cpp:32]: Do these have to live in global space? Or can they be local temps in the parseOptions function?

Comment (dlangbe) [reply test/llvm/hiptensor_options.cpp:32]: These options have to be globally accessible.

**Review by cgmillette — APPROVED** [review]
LGTM! Nice work

**Pull request merged**